### DISCUSSION TITLE
Implement a setting to enable document type assigning

### DISCUSSION THREAD

**Discussion by kireerik** [withastro/roadmap#588] [Proposal]
The default value of this setting can be `true` (This is the current behavior.).

My default recommendation for this value is `false`.

&nbsp;
> This proposal is based on @michrome's reply (https://github.com/withastro/roadmap/discussions/266#discussioncomment-5754502).

**Comment by kireerik** [comment]
I assume it is simple to implement this feature.

**Reply by kireerik** [reply]
@jasikpark What do you think about this?

**Comment by kireerik** [comment]
Eventually I preferred using [(static) Astro (page) (file) endpoints](https://docs.astro.build/en/core-concepts/endpoints/#static-file-endpoints) (instead of (Astro) components):
- # [refo](https://github.com/kireerik/refo)

&nbsp;
This completely eliminates this issue.

**Comment by matthewp** [comment]
Does this mean the `<doctype>`? It should already be the case that if you set the doctype yourself we do not set it. So if you use an old doctype other than HTML, that should work.

**Reply by michrome** [reply]
Hey @matthewp I am going to take a step back and describe my specific use case. I have a list of items that I want to pre-render as a snippet of HTML to inject into a `<ul>` using [htmx](https://htmx.org/).

So it’s an odd requirement but I effectively want a ‘broken’ page that does _not_ include `<doctype>`, `<html>` or `<body>`.

e.g. when I `curl` the URL for the page/snippet this is an example of the **complete response** I want:

```
<li>Item 4</li>
<li>Item 5</li>
<li>Item 6</li>
```

**Reply by kireerik** [reply]
Yes, that is the use case for example.

**Reply by michrome** [reply]
Hey @kireerik do you have a deep link (e.g. to a `.astro` file) where you’re doing this? I couldn’t see where you’re doing it in the repo above.  My understanding the approach would be more suited to returning JSON, not returning rendered HTML fragments/components.

**Reply by kireerik** [reply]
No I am not using Astro components anymore as I mentioned it here (https://github.com/withastro/roadmap/discussions/588#discussioncomment-6202400). I am using an [Astro](https://docs.astro.build/en/reference/cli-reference/#astro-dev) [Dynamic](https://docs.astro.build/en/core-concepts/routing/#static-ssg-mode) [Static file (server) endpoint](https://docs.astro.build/en/core-concepts/endpoints/#static-file-endpoints) to handle pages:
- [example/index/index/page.js](https://github.com/kireerik/refo/blob/0d08f7e3bcbd97cb2fdf1738493ed60e77b910de/example/index/index/page.js)

This way it is much better because I can set the document type myself if I want to:
- [example/index/index/site/main/index.jsx#L30](https://github.com/kireerik/refo/blob/0d08f7e3bcbd97cb2fdf1738493ed60e77b910de/example/index/index/site/main/index.jsx#L30)

**Reply by kireerik** [reply]
Maybe you can see the tools and services section (https://oengi.com/#toolsAndServices) on my website. That section is a separate path which is lazyloaded and added to the page. That is a path where I did not need the document type similarly to your use case.

**Reply by kireerik** [reply]
Luckily there is no such restriction. You can use them for anything.

**Reply by kireerik** [reply]
Yes, I have 1 [Astro](https://docs.astro.build/en/reference/cli-reference/#astro-dev) [Dynamic](https://docs.astro.build/en/core-concepts/routing/#static-ssg-mode) [Static file (server) endpoint](https://docs.astro.build/en/core-concepts/endpoints/#static-file-endpoints) at [example/index/index/index/pages/[...path].js](https://github.com/kireerik/refo/blob/0d08f7e3bcbd97cb2fdf1738493ed60e77b910de/example/index/index/index/pages/%5B...path%5D.js) defined in [example/index/index/page.js](https://github.com/kireerik/refo/blob/0d08f7e3bcbd97cb2fdf1738493ed60e77b910de/example/index/index/page.js) where the main rendering happens:
- [#L5](https://github.com/kireerik/refo/blob/0d08f7e3bcbd97cb2fdf1738493ed60e77b910de/example/index/index/page.js#L5)
- [#L126](https://github.com/kireerik/refo/blob/0d08f7e3bcbd97cb2fdf1738493ed60e77b910de/example/index/index/page.js#L126)

**Reply by kireerik** [reply]
@michrome Do you rely on Astro component features and if yes which ones? If not then you might be able to do something similar to what I did. Astro components have a lot of limitations for me and for me it is much better to use Astro without Astro components.

**Reply by michrome** [reply]
Thanks for clarifying!
